game/world/managers/objects/ai/BasicCreatureAI.py

```diff
@@ -22,6 +22,9 @@ def update_ai(self, elapsed):
         super().update_ai(elapsed)
         if not self.creature or not self.creature.combat_target:
             return
+        if self.creature.combat_target.get_type_id() == ObjectTypeIds.ID_PLAYER and self.creature.combat_target.beast_master:
+            self.creature.leave_combat()
```

Comment:
I think it's the caster who should be leaving combat, not the creature, no? Maybe check how the sanctuary handles the case so this can remove pending casts too.